.github/workflows/preview.yml

```diff
@@ -0,0 +1,129 @@
+          REACT_APP_SOLANA_API_URL: ${{ vars.REACT_APP_SOLANA_API_URL }}
+          REACT_APP_COVALENT_API_KEY: ${{ secrets.REACT_APP_COVALENT_API_KEY }}
+          REACT_APP_TRM_API_KEY: ${{ secrets.REACT_APP_TRM_API_KEY }}
+        run: |
+          pushd public
+          npm run set-version
+          popd
+          echo 'REACT_APP_VERSION=$npm_package_version' > .env
+          npm run build
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: preview-build
+          path: build
+      - name: Deploy metadata
+        run: |
+          mkdir deploy-metadata
+          pushd deploy-metadata
+          echo ${_BRANCH_NAME_SHA} > .branch-sha
+          echo ${_PKG_VERSION} > .version
+      - name: Upload Deploy metadata
+        uses: actions/upload-artifact@v3
+        with:
+          name: preview-deploy-metadata
+          path: deploy-metadata
+
+  publish:
+    name: "Publish"
+    needs: build
+    runs-on: "ubuntu-latest"
+    concurrency:
+      group: ${{ github.ref }}-preview
+      cancel-in-progress: true
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: preview-build
+      - name: Set up GitHub NPM registry
+        uses: actions/setup-node@v3
+        with:
+          node-version: v16.19.1
+          registry-url: https://npm.pkg.github.com
+      - name: Publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm publish
+  
+  deploy:
+    name: "Deploy Preview"
+    needs: publish
+    runs-on: "ubuntu-latest"
+    concurrency:
+      group: ${{ github.ref }}-preview
+      cancel-in-progress: true
+    permissions:
+      contents: write
+    environment:  Cloudflare-Preview
+    steps:
+      - name: Checkout preview branch
+        uses: actions/checkout@v3
+        with:
+          ref: preview
+      - name: Setup Git
+        run: |
+          git config user.name "xLabs CI"
+          git config user.email "devops@xlabs.xyz"
+      - name: Download Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: preview-deploy-metadata
+          path: deploy-metadata
+      - name: Set deploy metadata
+        run: |
+          echo "_PKG_VERSION=$(cat deploy-metadata/.version)" >> ${GITHUB_ENV}
+          echo "_BRANCH_NAME_SHA=$(cat deploy-metadata/.branch-sha)" >> ${GITHUB_ENV}
+      - name: Trigger a preview Deploy 
+        run: |
+          sleep 30s
+          mkdir -p preview/${{ env._BRANCH_NAME_SHA }}
+          pushd preview/${{ env._BRANCH_NAME_SHA }}
+          [[ ! -f .latest ]] && touch .latest
+          [[ ! -f .artifacts ]] && touch .artifacts
+          cat .latest >> .artifacts
+          echo ${{ env._PKG_VERSION }} > .latest
+          git add .artifacts .latest
+          git commit -m"Update preview artifacts ${{ env._PKG_VERSION }}"
```

Comment:
space missing here too

Reply:
Good eyes 👀 , but it does not affect tho :)